source/guides/clear/performance.rst

```diff
@@ -0,0 +1,236 @@
+software build. Patches include changes that default to more aggressive
+optimizations or optimizations that haven't yet been merged upstream.
+
+View the full list of patches in the autospec repositories on GitHub:
+
+* https://github.com/clearlinux-pkgs/gcc
+* https://github.com/clearlinux-pkgs/glibc
+* https://github.com/clearlinux-pkgs/llvm
+
+Aggressive compiler flags
+=========================
+
+|CL| uses aggressive
+`compiler flags <https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html>`_ to
+optimize software builds for runtime performance. Some significant flags that
+|CL| often implements are:
+
+`mtune and march <https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html>`_
+	Options used to tune generated code with optimized instructions for specific
+	CPU types instead of creating generic code for maximum compatibility.
+
+	|CL| defines it's minimum hardware requirements to be second-generation
+	Intel® microarchitecture code name Westmere (released in 2010) or later.
+	This enables the OS to build on newer CPU architecture features. Whenever
+	possible, |CL| tunes code for the Haswell generation processors or newer.
+
+	|CL| sets	:command:`march=westmere` and :command:`mtune=haswell`.
+
+	.. note::
+		|CL| doesn't require Advanced Encryption Standard (AES), so it may run
+		on some Intel CPUs from the first generation of Intel® microarchitecture
+		code name Nehalem (released in 2008). Refer to the
+		`recommended minimum system requirements <https://docs.01.org/clearlinux/latest/reference/system-requirements.html>`_ for specific requirements.
+
+`O3 <https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html>`_
+	The largest preset of compiler options optimizations for performance. O3
+	favors runtime performance.
+
+`LTO <https://gcc.gnu.org/onlinedocs/gccint/LTO.html>`_
+	Link-time optimization that performs an optimization between compiled object
+	files and creation of executable binaries by adding extra information to the
+	compiled object to help the linker.
+
+`PGO <https://en.wikipedia.org/wiki/Profile-guided_optimization>`_
+	Profile guided optimization or field guided optimization performs
+	optimization based on information sampled during the execution of the program.
+
+
+Compiler flags are set at different levels in the |CL| build environment:
+
+User flags
+	The set of default flags used by |CL| when a user compiles software
+	from source. The flags are exported as system-wide environment variables from
+	the
+	`/usr/share/defaults/etc/profile <https://github.com/clearlinux-pkgs/filesystem/blob/master/profile.x86_64>`_ file to the user’s shell by default. These are the
+	standard variables read by the compiler, named :command:`*FLAGS`, depending
+	on the compiler.
+
+	.. note::
+		Source code may come with software build systems that
+		override these values. This will cause a difference in expected flags.
+		The |CL| autospec tooling will ignore these overrides, but a manual

```

Comment:
... attempts to ignore these overrides... it's possible a build system still needs patching though.

Reply:
Wording corrected.